test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -33,7 +33,11 @@
     `(do ~@body)
     (let [case-versions (remove keyword? (take-nth 2 bindings))]
       `(doseq ~bindings
-         (testing (str "Testing case " '~case-versions)
+         (testing (str "Testing case: " (clojure.string/join
+                                          ", "
+                                          (map #(str %1 ": " %2)
+                                               '~case-versions
+                                               (list ~@case-versions))))
```

Comment:
Thanks, and seems like a good idea, though I might lean toward using normal let binding style for the values (produced via prn so there's no possibility of ambiguity), and making the prose chainable, since testing is intended to be nested.  So given
```clojure
(deftest some-test
  (testing "check something"
    (dotestseq [x [1 2 3]
                 y [:a :b]]
                (is (= 13 [x y])))))
```
we'd see something like this
```clojure
FAIL in (some-test) (core_test.clj:31)
check something with [[x 3] [y :b]]
expected: (= 13 [x y])
  actual: (not (= 13 [3 :b]))
```
given a definition like this
```clojure
(defmacro dotestseq [bindings & body]
  (if-not (seq bindings)
    `(do ~@body)
    (let [case-versions (remove keyword? (take-nth 2 bindings))]
      `(doseq ~bindings
         (testing (str "with " (pr-str (mapv vector
                                             '~case-versions
                                             ~(vec case-versions))))
           ~@body)))))
```
Though I'm not positive I got the expansion quite right, evaluation-wise.